static/js/components/PersonalTab.js

```diff
@@ -68,7 +68,15 @@ export default class PersonalTab extends React.Component {
     dispatch(setProfileStep(PERSONAL_STEP));
   }
 
-  onChange = (selection: Option): void => {
+  componentDidUpdate() {
+    const { currentProgramEnrollment } = this.props;
+    const selectedProgram = this.getSelectedProgramId();
+    if ( currentProgramEnrollment && !selectedProgram ) {
+      this.onProgramSelect({ value: currentProgramEnrollment.id, label: '' });
```

Comment:
yeah, it happens to do what we want, but it smells. the fact that we're sort of cheating and creating a fake `Option` with an empty `label` value is one indicator of that. at this point in execution, all we want to do is make sure we have a selected program set in the state. the concept of an Option with a label and a value is irrelevant. i think we should split `onProgramSelect` like so:

```javascript
setProgramById= (programId: number?): void => {
    const {
      programs,
      setProgram,
    } = this.props;
    if ( programId ) {
      let selected = programs.find(program => program.id === parseInt(programId));
      setProgram(selected);
    } else {
      setProgram(null);
    }
}

onProgramSelect = (selection: Option): void => {
    let programId = selection && selection.value ? parseInt(selection.value) : null;
    this.setProgramById(programId);
}
```

`setProgramById` might be relevant to other components too. i feel like it would be better if we just called the action with a simple program id in the argument (instead of finding the right object in the `programs` list) but that's a bit more work, perhaps for a different PR